hledger-lib/Hledger/Reports/MultiBalanceReport.hs

```diff
@@ -558,15 +558,16 @@ cumulativeSum value start = snd . M.mapAccumWithKey accumValued start
 postingAndAccountValuations :: ReportSpec -> Journal -> PriceOracle
                             -> (DateSpan -> Posting -> Posting, DateSpan -> Account -> Account)
 postingAndAccountValuations ReportSpec{rsToday=today, rsOpts=ropts} j priceoracle = case value_ ropts of
-    Just (AtEnd _) -> (const id, avalue' (cost_ ropts) (value_ ropts))
-    _              -> (pvalue' (cost_ ropts) (value_ ropts), const id)
+    Just v@(AtEnd _) -> (pvalue' (cost_ ropts) Nothing, avalue' v)
+    _                -> (pvalue' (cost_ ropts) (value_ ropts), const id)
   where
-    avalue' c v span a = a{aibalance = value (aibalance a), aebalance = value (aebalance a)}
-      where value = mixedAmountApplyCostValuation priceoracle styles (end span) today (error "multiBalanceReport: did not expect amount valuation to be called ") c v  -- PARTIAL: should not happen
-    pvalue' c v span = postingApplyCostValuation priceoracle styles (end span) today c v
-    end = fromMaybe (error "multiBalanceReport: expected all spans to have an end date")  -- XXX should not happen
-        . fmap (addDays (-1)) . spanEnd
+    pvalue' c v span = maybeStripPrices . postingApplyCostValuation priceoracle styles (end span) today c v
```

Comment:
I feel this function is getting harder to understand for code readers. Could you try to clarify what's going on with a comment - why we might be stripping prices here, eg. Feel free to mention issue numbers if you think it will help.

Reply:
Sure.